containers/src/Kokkos_DynRankView.hpp

```diff
@@ -921,9 +929,14 @@ KOKKOS_INLINE_FUNCTION auto subdynrankview(
                     (drv.rank() > 4 && !std::is_integral_v<SubArg4> ? 1 : 0) +
                     (drv.rank() > 5 && !std::is_integral_v<SubArg5> ? 1 : 0) +
                     (drv.rank() > 6 && !std::is_integral_v<SubArg6> ? 1 : 0);
-  return DynRankView<typename sub_t::value_type, typename sub_t::array_layout,
-                     typename sub_t::device_type,
-                     typename sub_t::memory_traits>(sub, new_rank);
+
+  using return_type =
+      DynRankView<typename sub_t::value_type, Kokkos::LayoutStride,
+                  typename sub_t::device_type, typename sub_t::memory_traits>;
+  return static_cast<return_type>(
+      DynRankView<typename sub_t::value_type, typename sub_t::array_layout,
+                  typename sub_t::device_type, typename sub_t::memory_traits>(
+          sub, new_rank));
```

Comment:
Why did you need to change `return_type`?

Reply:
Because DynRankView used to only ever return LayoutStride in subviews, and some code downstream was relying on that (doing ternary expression with different dimensionality subview, which resulted in different types).

Reply:
If that's something to change later again (after refactoring Sacado) we should add a comment to that extent.

Reply:
Its not actually Sacado its in Intrepid2

Reply:
We can make that part of the DynRank Review later.